sqlserver/tests/test_statements.py

```diff
@@ -327,7 +328,7 @@ def test_statement_metrics_and_plans(
     # dbm-metrics
     dbm_metrics = aggregator.get_event_platform_events("dbm-metrics")
     assert len(dbm_metrics) == 1, "should have collected exactly one dbm-metrics payload"
-    payload = dbm_metrics[0]
+    payload = next((n for n in dbm_metrics if n.get('collection_type') == 'query_metrics'), None)
```

Comment:
do we need a new field on our events called `collection_type` could we differentiate via `dbm_type` or nah? 

I'm not sure at the moment if that would require more extensive backend changes or not